pkg/services/ssosettings/database/database_test.go

```diff
@@ -80,45 +80,120 @@ func TestIntegrationUpsertSSOSettings(t *testing.T) {
 	t.Run("insert a new SSO setting successfully", func(t *testing.T) {
 		setup()
 
-		expected := &models.SSOSetting{
-			Provider: "azuread",
-			Settings: map[string]interface{}{
-				"enabled": true,
-			},
+		provider := "azuread"
+		settings := map[string]interface{}{
+			"enabled":   true,
+			"client_id": "azuread-client",
 		}
 
-		err := ssoSettingsStore.Upsert(context.Background(), "azuread", map[string]interface{}{
-			"enabled": true,
-		})
+		err := ssoSettingsStore.Upsert(context.Background(), provider, settings)
 		require.NoError(t, err)
 
-		actual, err := ssoSettingsStore.Get(context.Background(), "azuread")
+		actual, err := getSSOSettingsByProvider(sqlStore, provider, false)
 		require.NoError(t, err)
+		require.Equal(t, settings, actual.Settings)
+		require.WithinDuration(t, time.Now(), actual.Created, 5*time.Minute)
+		require.WithinDuration(t, time.Now(), actual.Updated, 5*time.Minute)
 
-		require.True(t, maps.Equal(expected.Settings, actual.Settings))
+		deleted, notDeleted, err := getSSOSettingsCountByDeleted(sqlStore)
+		require.NoError(t, err)
+		require.EqualValues(t, 0, deleted)
+		require.EqualValues(t, 1, notDeleted)
 	})
 
 	t.Run("replaces an existing SSO setting for the specified provider", func(t *testing.T) {
 		setup()
 
-		err := ssoSettingsStore.Upsert(context.Background(), "azuread", map[string]interface{}{
-			"enabled": true,
-		})
+		provider := "github"
```

Comment:
nit: I think it's a bit confusing that provider is `github` but `client_id` refers to azure.

```suggestion
		provider := "azuread"
```

Reply:
Good catch! I'll fix it.

Reply:
I updated the client id instead to have some diversification.